pkg/universe.dagger.io/x/nico@meisenzahl.org/terraform/init.cue

```diff
@@ -0,0 +1,55 @@
+// Deploy with Terraform
+// https://www.terraform.io
+
+// TODO:
+// - Add support for injecting generic backend configuration secrets
+
+package terraform
+
+import (
+	"dagger.io/dagger"
+	"dagger.io/dagger/core"
+
+	"universe.dagger.io/docker"
+)
+
+// Terraform init
+#Init: {
+	// Terraform version
+	version: string | *"latest"
+
+	// Source code
+	source: dagger.#FS
+
+	_run: docker.#Build & {
+		steps: [
+			docker.#Pull & {
+				source: "hashicorp/terraform:\(version)"
+			},
+
+			// Copy source
+			docker.#Copy & {
+				dest:     "/src"
+				contents: source
+			},
+
+			// Run init
+			docker.#Run & {
+				command: {
+					name: "init"
+					args: ["-input=false", "-upgrade"]
+				}
+				workdir: "/src"
+			},
+		]
+	}
+
+	// Output source and .terraform folder
+	_output: core.#Subdir & {
+		input: _run.output.rootfs
+		path:  "/src/"
+	}
```

Comment:
He's not using `export` because of `docker.#Build`, but if the copy is a mount instead, then you can reduce all of this into two steps: 1) pull image; 2) run command with mount and export at the same time.